UnityProject/Assets/Scripts/Systems/ChemistryEffects/ChemExplosionPullPush.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections;
+using UnityEngine;
+using Systems.Explosions;
+using HealthV2;
+using Core;
+using Cysharp.Threading.Tasks;
+using Logs;
+using Systems.Atmospherics;
+using TileManagement;
+using UniversalObjectPhysics = Core.Physics.UniversalObjectPhysics;
+
+namespace Chemistry.Effects
+{
+	[CreateAssetMenu(fileName = "effect", menuName = "ScriptableObjects/Chemistry/Effect/ChemExplosionPullPush")]
+	public class ChemExplosionPullPush : ChemExplosion
+	{
+		[SerializeField, Tooltip("Should this effects force push not pull")] private bool ShouldPush = false;
+		[SerializeField] private string effectName = "Dark Matter";
+
+
+		[SerializeField] private GameObject cataclysmPrefab = null;
+		[SerializeField] private float cataclysmThreshold = 2000;
+
+		public override IEnumerator NowExplosion(MonoBehaviour sender, float amount)
+		{
+			yield return WaitFor.Seconds(Delay);
+
+			UniversalObjectPhysics objectBehaviour = sender.GetComponentCustom<UniversalObjectPhysics>();
+			RegisterObject registerObject = sender.GetComponentCustom<RegisterObject>();
+			BodyPart bodyPart = sender.GetComponentCustom<BodyPart>();
+			ExplosionNode node = ExplosionTypes.NodeTypes[explosionType];
+
+			float strength = ChemistryUtils.CalculateYieldFromReaction(amount, potency);
+
+			bool insideBody = bodyPart == true && bodyPart.HealthMaster == true;
+
+			var picked = sender.GetComponentCustom<Pickupable>();
+			if (picked != null)
+			{
+				//If sender is in an inventory use the position of the inventory.
+				if (picked.ItemSlot != null)
+				{
+					objectBehaviour = picked.ItemSlot.ItemStorage.GetRootStorageOrPlayer().GetComponentCustom<UniversalObjectPhysics>();
```

Comment:
picked.ItemSlot.game object.GetRootGameObject should theoretically work and be a bit cleaner